### ISSUE TITLE
Error when tagging with BidLSTM_CNN architecture

### ISSUE THREAD

**Issue by Protossnam** [kermitt2/delft#28]
My warm thanks for your wonderful works! I have 2 OS (Windows and Ubuntu), both are using Python 3.6 (with Anaconda 3). I installed tensorflow-gpu version 1.10. 

I've successfully trained BidLSTM_CNN model and obtained the `model_weights.hdf5` file. 

When I run: `python nerTagger.py --dataset-type conll2003 --architecture BidLSTM_CNN eval`

I get:
```
Evaluation on test set:
        f1 (micro): 89.38
                  precision    recall  f1-score   support

            MISC     0.7974    0.7735    0.7852       702
             ORG     0.8341    0.8958    0.8639      1661
             LOC     0.9052    0.9269    0.9159      1668
             PER     0.9465    0.9511    0.9488      1617

all (micro avg.)     0.8822    0.9056    0.8938      5648

runtime: 5.412 seconds
```

However, when I try to perform action tag (instead of eval), it gives me the same error on both 2 OS (NOTE: Here, I copied the result from Windows terminal. By the way, on Ubuntu terminal, it also said the same thing):

```
Traceback (most recent call last):
  File "nerTagger.py", line 464, in <module>
    file_out=file_out)
  File "nerTagger.py", line 381, in annotate
    model.tag_file(file_in=file_in, output_format=output_format, file_out=file_out)
  File "D:\Anaconda3\envs\ULR\delft-master\delft\sequenceLabelling\wrapper.py", line 275, in tag_file
    annotations = tagger.tag(texts, output_format)
  File "D:\Anaconda3\envs\ULR\delft-master\delft\sequenceLabelling\tagger.py", line 86, in tag
    piece["entities"] = self._build_json_response(tokens, tags, prob, offsets)["entities"]
  File "D:\Anaconda3\envs\ULR\delft-master\delft\sequenceLabelling\tagger.py", line 113, in _build_json_response
    chunks = get_entities_with_offsets(tags, offsets)
  File "D:\Anaconda3\envs\ULR\delft-master\delft\sequenceLabelling\tagger.py", line 156, in get_entities_with_offsets
    end_pos = offsets[j-1][1]-1
IndexError: list index out of range
```

**Comment by kermitt2** [comment]
Hello @Protossnam ! Thank you for reporting the issue. 
I am not able to reproduce the issue, which input file are you using for tagging?

```
(env) lopez@trainer:~/delft$ python nerTagger.py --dataset-type conll2003 --architecture BidLSTM_CNN --file-in data/test/test.ner.en.txt tag
/home/lopez/delft/env/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.
data/test/test.ner.en.txt
loading model weights data/models/sequenceLabelling/ner-en-conll2003-BidLSTM_CNN/model_weights.hdf5
{
    "software": "DeLFT",
    "date": "2019-04-25T21:42:10.310022",
    "model": "ner-en-conll2003-BidLSTM_CNN",
    "texts": [
        {
            "text": "The University of California has found that 40 percent of its students suffer food insecurity. At four state universities in Illinois, that number is 35 percent.",
            "entities": [
                {
                    "text": "University of California",
                    "class": "ORG",
                    "score": 0.8168373703956604,
                    "beginOffset": 4,
                    "endOffset": 27
                },
                {
                    "text": "Illinois",
                    "class": "LOC",
                    "score": 0.9999653100967407,
                    "beginOffset": 125,
                    "endOffset": 132
                }
            ]
        },
        {
            "text": "President Obama is not speaking anymore from the White House.",
            "entities": [
                {
                    "text": "Obama",
                    "class": "PER",
                    "score": 0.9876206517219543,
                    "beginOffset": 10,
                    "endOffset": 14
                },
                {
                    "text": "White House",
                    "class": "LOC",
                    "score": 0.9870116114616394,
                    "beginOffset": 49,
                    "endOffset": 59
                }
            ]
        },
...
```

**Comment by Protossnam** [comment]
Thank @kermitt2 for your reply. I use this text file for tagging:
[CoNNL-EN-TEST.txt](https://github.com/kermitt2/delft/files/3120128/CoNNL-EN-TEST.txt)

With BidLSTM_CRF and BidLSTM_CNN_CRF architecture, I'm able to retrieve the output JSON file. Here is the example for BidLSTM_CRF (you can change the file name extensions from .txt to .json):
[test_BidLSTM_CRF.txt](https://github.com/kermitt2/delft/files/3120141/test_BidLSTM_CRF.txt)

**Comment by kermitt2** [comment]
Thank you very much for the details! 

In the specific BidLSTM_CNN architecture masking zero and the last TimeDistributed layer (which was useless) created a padding problem, the consequence was that the number of tokens and the number of labels were not identical. 

Everything should be fixed with commit b76643b5d39115f649a8d258bc746924dd7767de

**Comment by Protossnam** [comment]
That's it ;) Thank you for your wonderful work again <3 I'm going to close this issue @kermitt2
